src/lib/protocols/ssh.c

```diff
@@ -57,43 +62,95 @@
 /* #define SSH_DEBUG 1 */
 
 static void ndpi_search_ssh_tcp(struct ndpi_detection_module_struct *ndpi_struct, struct ndpi_flow_struct *flow);
-  
+
 /* ************************************************************************ */
 
-static void ssh_analyse_signature_version(struct ndpi_detection_module_struct *ndpi_struct,
+static void ssh_analyze_signature_version(struct ndpi_detection_module_struct *ndpi_struct,
 					  struct ndpi_flow_struct *flow,
 					  char *str_to_check,
 					  u_int8_t is_client_signature) {
 
+  if (str_to_check == NULL) return;
+
+  int major = 0;
+  int minor = 0;
+  int patch = 0;
+  int obsolete_ssh_version = 0;
+
+  if (sscanf(str_to_check, "SSH-%*f-OpenSSH_%d.%d.%d", &major, &minor, &patch) != 3) 
```

Comment:
Did you search fo other SSH implementations other than OpenSSH? Please do

Reply:
FYI: https://en.m.wikipedia.org/wiki/Comparison_of_SSH_servers

It would be very nice to have at least additional support for dropbear SSH as that implementation is used in embedded environments.